### PR TITLE
ID switch for some missed detection reappear

### PR DESCRIPTION
**Pull request by nikk-nikaznan**
Idea for ID switch, doesn't need to merge and can be closed if it's not what we want
- Add last known id for predicted ID that not nan
- For crabs appear, if gt_id is in last_known_id, we check if the current_predicted_id has changed

### CODE REVIEW

**Review by sfmig — APPROVED** [review]
thanks Nik for implementing this idea! 🚀 

Mainly one question about an `if gt_to_tracked_id_previous_frame is None` case, the rest are small suggestions.

Would it be annoying to add a simple test to cover this extra case? If it is feel free to open an issue and I will deal with it.

I _think_ in the HOTA paper they select a 'permanent' mapping from GT to predicted ID such that the MOTA metric is as high as possible (whereas here we are defining a mapping based on the last assigned value). I think this is fine and clear for now, but maybe something to consider for the future... 🤔

File: crabs/tracker/evaluate_tracker.py [hunk 1]
```diff
@@ -168,11 +170,14 @@ def count_identity_switches(
         int
             The number of identity switches between the two sets of object IDs.
         """
+
         if gt_to_tracked_id_previous_frame is None:
+            for gt_id, pred_id in gt_to_tracked_id_current_frame.items():
```
[the remark this answers is not in the document]
Comment (nikk-nikaznan) [reply crabs/tracker/evaluate_tracker.py:175]: So this is for the first frame. We will not have any previous frame but we want to save the last known predicted ID for later for the current frame.
So let's say the first frame has predicted ID, second missed, and third frame has predicted ID, we have the value of the last known ID.

[the remark this answers is not in the document]
Comment (sfmig) [reply crabs/tracker/evaluate_tracker.py:175]: aaah sorry I misread! all good!

**Comment by nikk-nikaznan** [comment]
Thank you Sofia! I just add some test, let me know what you think.
I realised as well when debug for the test, that we need to filter out the gt_id for pred_id of np.nan. Otherwise, all gt_id will be in the continuous one right.

**Comment by sfmig** [comment]
thanks Nik, the tests look great! ⭐ 

And I think you picked up something we had missed with this point:
> we need to filter out the gt_id for pred_id of np.nan.

I think we actually don't need to do this.... I think the confusion comes because we were considering that an 'historical' ID switch (i.e. ID switch that is computed using info in `last_known_predicted_ids`) can only happen if a crab appears (according to the ground truth). However, It can also happen if a crab continues to exist.

I edited the code to add this second case and added two more tests. Let me know what you think! 

If you give me your thumbs up, feel free to merge and I'll get the README for id switches wrapped up.

**Pull request merged**